static/js/components/google_map.tsx

```diff
@@ -0,0 +1,257 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Component for embedding a map from Google Maps
+ */
+
+import axios from "axios";
+import _ from "lodash";
+import React from "react";
+
+const DEFAULT_MAP_ZOOM = 4;
+
+// number of digits after decimal point in LatLong/<lat>_<long> dcids
+const LAT_LONG_PRECISION = 5;
+
+interface GoogleMapPropType {
+  /**
+   * DCID of the place/event to show a map for.
+   */
+  dcid: string;
+}
+
+interface GoogleMapStateType {
+  /**
+   * Stores responses from api calls.
+   */
+  markerLocation: GoogleMapCoordinates;
+  mapInfo: MapInfoResponse;
+}
+
+interface GoogleMapCoordinates {
+  /**
+   * Lat/Long coordinates, stored in a form the Google maps API expects.
+   */
+  lat: number;
+  lng: number;
+}
+
+interface MapInfoResponse {
+  /**
+   * Response format of /api/place/mapinfo/.
+   */
+  up: number;
+  down: number;
+  left: number;
+  right: number;
+  coordinateSequenceSet: google.maps.LatLng[][];
+}
+
+/**
+ * Converts a DCID of the form latLong/<latitude>_<longitude> to
+ * lat/long coordinates used by the google maps API.
+ * @param dcid latLong DCID to convert
+ * @returns a {lat: latitude, lng: longitude} object for the google maps API
+ */
+function convertLatLongDcid(dcid: string): GoogleMapCoordinates {
+  if (dcid.startsWith("latLong/") && dcid.includes("_")) {
+    const coordinateStrings = dcid.slice("latLong/".length).split("_");
+    const coords = coordinateStrings.map((coordString) =>
+      // Add decimal point to parsed latitude and longitude numbers
+      parseFloat(
+        coordString.slice(0, -LAT_LONG_PRECISION) +
+          "." +
+          coordString.slice(-LAT_LONG_PRECISION)
+      )
+    );
+    return { lat: coords[0], lng: coords[1] };
+  }
+  return null;
+}
+
+/**
+ * Initialize a google map widget.
+ * @param container the element to render the map in
+ */
+function initMap(container: HTMLDivElement): google.maps.Map {
+  // set height for map to be visible
+  container.style.height = "100%";
+  container.style.minHeight = "160px";
+  const mapOptions = {
+    mapTypeControl: false,
+    draggable: true,
+    scaleControl: true,
+    scrollwheel: true,
+    navigationControl: true,
+    streetViewControl: false,
+    mapTypeId: google.maps.MapTypeId.ROADMAP,
+    zoom: DEFAULT_MAP_ZOOM,
+  };
+  return new google.maps.Map(container, mapOptions);
+}
+
+/**
+ * Draw polygons on map given KML coordinates for each polygon.
+ * @param state current component state with KML Coordinates stored in it
+ * @param map map to draw polygon in
+ */
+function drawKmlCoordinates(
+  state: GoogleMapStateType,
```

Comment:
nit: instead of taking the whole state as a param, just need `mapInfo: MapInfoResponse`

Reply:
Good catch! Updated.